### DISCUSSION TITLE
Modification of tempo with web interface

### DISCUSSION THREAD

**Discussion by theblackhat17** [onlaj/Piano-LED-Visualizer#323] [Q&A]
Hey, I have just configured the entire piano project, everything works perfectly I am delighted!
Today, I would like to add a functionality, it is to modify the tempo of a piano piece using the web interface directly. Indeed, we can do this from the small OLED screen, so I would like to know if I can retrieve the variable from the python program to call it in the html page and let the user modify his piano tempo to play more or less fast. 
I will add the speed modes in the web interface directly with the attributes : slow(25%) / medium(50%) / fast(75%) / hard(100%). All this by changing the tempo percentage which is 100% at the beginning.
I saw that there was the function "**set_tempo**" or "**song_tempo**" present in the file "_learnmidi.py_".

`self.set_tempo = 100` or `self.song_tempo = 500000` or `self.ticks_per_beat = 240` 

In my new page named "songspead.html" and added in the menu, I added the four options in the list like this :                      `<option value="0">Slow</option> <option value="0">Medium</option> <option value="0">Fast</option> <option value="0">Hard</option> `

Do you think it would be possible?
Thanks for your help!

Th3_bl@ck_h@t

**Comment by jwessel** [comment]
Personally, I thought it would be nice to have a learn button added to the song playback screen. 

Having a separate learn tab in the web interface also seems like it would be nice and should be entirely possible because the LCD menu already has all the functions implemented that you would need to call from the Web interface.

**Reply by theblackhat17** [reply]
Thank you very much for your reply @jwessel, the problem I'm having now after having the same thought as you is that I don't know how to call this "python" function in my HTML file. Do you have any resource documents? 
Thanks to you
